pytorch_lightning/plugins/training_type/ddp.py

```diff
@@ -383,16 +393,28 @@ def reduce(self, tensor, group: Optional[Any] = None, reduce_op: Union[ReduceOp,
         return tensor
 
     def training_step(self, *args, **kwargs):
-        return self.model(*args, **kwargs)
+        if isinstance(self.model, DistributedDataParallel):
+            return self.model(*args, **kwargs)
+        else:
+            return self.model.training_step(*args, **kwargs)
```

Comment:
during training, the first case should always be True
i'm not sure if we ought to assert this every step though.
```suggestion
        return self.model(*args, **kwargs)
```

Reply:
yes. you are right. Initially, I want to make all the *_step's have the consistent way to forward, but I think it's not necessary for training_step.